### ISSUE TITLE
v0.8.1 breaks post previews

### ISSUE THREAD

**Issue by jb510** [Automattic/Edit-Flow#344]
Not sure why yet, but post previews are failing on v0.8.1. 

I rolled back to 0.7.6, still failing, rolled up to 0.8.0 and they started working 😕 

I manually installed 0.8.1, again broken, went back to 0.8.0 for now and it's working fine 😖 

This is the primary install on a multisite (just a couple sub-sites).  I've recreated the issue on staging site as well.  We use CPTs and PFs, but this is just a standard post.

**Comment by cojennin** [comment]
Having some trouble recreating this on my setup (Edit Flow v0.8.1, WordPress v4.5). 

Quick questions:
What version of WordPress are you using?
What kind of permalink structure do you have?
How are you previewing it? (hitting the preview button on the Post screen, hitting the preview link from the page listing all posts, etc)

Something quick to try:
After you install Edit Flow v0.8.1, go to the "Permalinks" settings page and click the "Save Changes" button. See if that helps with previewing.

**Comment by jb510** [comment]
Thanks for trying to reproduce.

WP Version: 4.5 and 4.5.1
Permalinks: Day and name /2016/05/02/sample-post/
PHP: 5.6 (WP Engine)
Steps: Same behavior from edit post page (preview on post screen) as post list (hover, quick links)

I deactivated all plugins and it's still occurring...  TwentySixteen, still occurring.

I can debug line by line locally later.

**Comment by cojennin** [comment]
Following up for some more info:

1. Just to make sure, when you say "failing" you mean they're throwing a 404? Or some other kind of failure
2. When you deactivate Edit Flow previewing starts working again?
3. When you activate Edit Flow but no other plugins previewing stops working?

**Comment by jb510** [comment]
1. yes, WP 404
2. yes, deactivated or running 0.8.0 preview works, running 0.8.1 previews do not
3. yes, with _only_ editflow 0.8.1 active and running TwentySixteen, previews go to a 404

It would be great to get confirmation, but this isn't urgent. I'll debug later this week.

**Comment by cojennin** [comment]
Hrm. Yea, having trouble trying to get this to reproduce. Haven't seen other reported instances of this issue, but if I do I'll tag them here.

Let me know what you find when you have some time to debug.

**Comment by cojennin** [comment]
@jb510 did you have some time to dig into what might be causing this?

**Comment by jb510** [comment]
not yet.

**Comment by cojennin** [comment]
@jb510 quick question, is it the link in the screenshot below that's returning a 404 when clicked?

![screen shot 2016-05-12 at 3 37 29 pm](https://cloud.githubusercontent.com/assets/1888152/15227870/3b833cda-1857-11e6-8540-eb54f575cc82.png)

**Comment by jb510** [comment]
No.  Not `view post` permalink.  The `preview post` button in the publish dialog box.
<img width="297" alt="2016-05-12 at 2 49 pm" src="https://cloud.githubusercontent.com/assets/801241/15234766/bfc0d6d0-1850-11e6-872c-54273b2126ec.png">

also happens using preview link from post list view:
<img width="180" alt="2016-05-12 at 2 48 pm" src="https://cloud.githubusercontent.com/assets/801241/15234774/cf3bb986-1850-11e6-845b-d805872a816e.png">

**Comment by philipjohn** [comment]
@jb510 What's the URL you're taken too after clicking the Preview button?

**Comment by jb510** [comment]
From the post list (hover quick links):
URL is on hover/click is:
/?p=68807&preview=true
however it immediately redirects to just
/?p=68807

Which shows a WP 404

It's strange, there is no nonce.

From the post edit (preview button)
URL is on hover/click is:
/?p=68807&preview=true
however it immediately redirects to just
/?p=68807&preview_id=68807&preview_nonce=102b742e81&post_format=image

Which again shows a WP 404

This happens on production and staging.  on Schedule and draft posts. Preview does work for published posts.  

When Edit Flow is deactivated previews for scheduled/draft posts work normally

Made a screencast, if the video/audio doesn't work let me know as I've been having problems with videos on CloudUp lately: https://cloudup.com/cK7HpR4Z84U

**Comment by philipjohn** [comment]
Does this also happen for you in a vanilla WP install with nothing else active? I'm not able to reproduce so I'm wondering if it's more than just Edit Flow we need to consider.

**Comment by jeremyfelt** [comment]
We ran into what I think is the same issue that @jb510 describes. I've submitted #357 as a fix.

In a multisite subdirectory configuration, the preview URL `http://wp.wsu.dev/sitename?page_id=1` will result in a page lookup at the `wp.wsu.dev` site rather than a preview on the `wp.wsu.dev/sitename/` site. Adding the trailing slash via `home_url( '/' )` is consistent with how the preview URLs are built in core.
